driver/src/orderbook/streamed/state.rs

```diff
@@ -0,0 +1,531 @@
+use super::*;
+use crate::contracts::{
+    stablex_auction_element,
+    stablex_contract::batch_exchange::{event_data::*, Event},
+};
+use crate::models::Order as ModelOrder;
+use ethcontract::U256;
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::iter::Iterator;
+use thiserror::Error;
+
+// TODO: Should we handle https://github.com/gnosis/dex-contracts/issues/620 ?
+// There is a workaround detailed in the issue it hasn't been implemented.
+
+// Most types, fields, functions in this module mirror the smart contract because we need to
+// emulate what it does based on the events it emits.
```

Comment:
I was thinking that this isn't relevant to users of the struct (and module) because it is about implementation details. It is only developers editing this file. Not that I mind either way but a normal comment felt more appropriate.